### DISCUSSION TITLE
Variable to access title of a pdf tiddler while clipping to a seperate notes tiddler.

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#111] [Q&A]
Sometimes I take notes from embedded pdf tiddlers in my wiki using the `pdfclipconfig` rules. Now I am trying to use the `Tidclipconfig` to clip text and snaps from embedded pdf tiddlers into another separate tiddler in which I am taking notes. So I will press the dock viewtoolbar button of the notes tiddler and then try to clip from the pdf tiddler. For this to work, I need to have a variable which has access to the title of the pdf tiddler (of the format `pdfname.pdf` ). Is there any variable available which can access the title of the pdf tiddler. I can't use the `@section` since I am not docking to the pdf tiddler, but to the notes tiddler.

https://demo-complete.tiddlyhost.com/#

**Comment by buggyj** [comment]
There is no variable for this. I would suggest copying the title of the pdf tiddler to the clipboard and then using `@clip` to clip the title.

**Reply by arunnbabu81** [reply]
> There is no variable for this. I would suggest copying the title of the pdf tiddler to the clipboard and then using `@clip` to clip the title.


I can do that for snip. But for snaps it won't work because clipsnap will conflict with clip.


Can there new variable or viewtoolbar button to clip to particular tiddler like the localtcplug in

**Comment by buggyj** [comment]
In the tiddlywiki which contains the `some.pdf` tiddler, you could permalink that tiddler and the name will appear in the url - `/tiddlywiki.html#some.pdf` and extract the tiddler title from the `@pageRef` variable.

**Reply by arunnbabu81** [reply]
> In the tiddlywiki which contains the `some.pdf` tiddler, you could permalink that tiddler and the name will appear in the url - `/tiddlywiki.html#some.pdf` and extract the tiddler title from the `@pageRef` variable.

Thats a good option. Can you tell how to extract the tiddler title from @pageRef using regexp.

**Comment by buggyj** [comment]
Try this

```
#rule=/[\s\S]*#(.*)/g
#term=$1
#tidtitle=((*@pageRef/#rule/#term*))
```

**Reply by arunnbabu81** [reply]
> Try this
> 
> ```
> #rule=/[\s\S]*#(.*)/g
> #term=$1
> #tidtitle=((*@pageRef/#rule/#term*))
> ```

This works except when there is space in the tiddler title. `%20` gets added in the extracted tiddler title instead of the `space`. Is there any way to remove this?

**Reply by buggyj** [reply]
Try adding
```
#rule2=/%20/g
#term2= 
#tidtitle=((*#tidtitle/#rule2/#term2*))
```

**Reply by arunnbabu81** [reply]
> Try adding
> 
> ```
> #rule2=/%20/g
> #term2= 
> #tidtitle=((*#tidtitle/#rule2/#term2*))
> ```

This one works. Thank you
